src/main/java/org/openrewrite/staticanalysis/RemoveUnusedParams.java

```diff
@@ -0,0 +1,182 @@
+                        return m.getMethodType() != null && m.getMethodType().isOverride();
+                    }
+
+                    private boolean skipIfOverriddenElsewhere(String signature) {
+                        return acc.OVERRIDE_SIGNATURES.contains(signature);
+                    }
+
+                    @Override
+                    public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                        J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
+
+                        if (skipAnyExplicitOverride(m)) {
+                            return m;
+                        }
+
+                        if (skipIfOverriddenElsewhere(buildSignature(m))) {
+                            return m;
+                        }
+
+                        if (m.getBody() == null ||
+                                m.hasModifier(J.Modifier.Type.Native) ||
+                                !m.getLeadingAnnotations().isEmpty()) {
+                            return m;
+                        }
+
+                        Set<String> params = m.getParameters().stream()
+                                .filter(p -> p instanceof J.VariableDeclarations)
+                                .flatMap(p -> ((J.VariableDeclarations) p).getVariables().stream())
+                                .map(J.VariableDeclarations.NamedVariable::getSimpleName)
+                                .collect(Collectors.toSet());
+
+                        Set<String> used = new HashSet<>();
+                        new JavaIsoVisitor<Set<String>>() {
+                            Deque<Set<String>> shadowed = new ArrayDeque<>();
+
+                            @Override
+                            public J.Block visitBlock(J.Block block, Set<String> u) {
+                                shadowed.push(new HashSet<>());
+                                J.Block b = super.visitBlock(block, u);
+                                shadowed.pop();
+                                return b;
+                            }
+
+                            @Override
+                            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations vars, Set<String> u) {
+                                vars.getVariables().forEach(v -> shadowed.peek().add(v.getSimpleName()));
+                                return super.visitVariableDeclarations(vars, u);
+                            }
+
+                            @Override
+                            public J.Identifier visitIdentifier(J.Identifier ident, Set<String> u) {
+                                for (Set<String> scope : shadowed) {
+                                    if (scope.contains(ident.getSimpleName())) {
+                                        return ident;
+                                    }
+                                }
+                                if (params.contains(ident.getSimpleName())) {
```

Comment:
```suggestion
                                if (m.getParameters().stream() .filter(p -> p instanceof J.VariableDeclarations) .flatMap(p -> ((J.VariableDeclarations) p).getVariables().stream()) .map(J.VariableDeclarations.NamedVariable::getSimpleName) .collect(Collectors.toSet()).contains(ident.getSimpleName())) {
```
extract this in dedicated 1LOC method giving good long name hide impl. detail away which is now done with high coupling and low cohesion over field.

try inline one-time usage fields always as most only coupling.

https://checkstyle.sourceforge.io/checks/coding/variabledeclarationusagedistance.html


```java
                                if (isFoo())) {
```

Reply:
Addressed.